src/elements/common/sub-header/SubHeaderRight.js

```diff
@@ -7,35 +7,46 @@
 import React from 'react';
 import Sort from './Sort';
 import Add from './Add';
+import GridViewButton from './GridViewButton';
 import { VIEW_FOLDER } from '../../../constants';
+import { FeatureFlag } from '../feature-checking';
 import './SubHeaderRight.scss';
 
 type Props = {
     canCreateNewFolder: boolean,
     canUpload: boolean,
     currentCollection: Collection,
+    isGridView: boolean,
     onCreate: Function,
+    onGridViewSwitch?: Function,
     onSortChange: Function,
     onUpload: Function,
     view: View,
 };
 
 const SubHeaderRight = ({
-    view,
-    onUpload,
-    onCreate,
-    canUpload,
     canCreateNewFolder,
+    canUpload,
     currentCollection,
+    isGridView,
+    onCreate,
+    onGridViewSwitch,
     onSortChange,
+    onUpload,
+    view,
 }: Props) => {
     const { sortBy, sortDirection, items = [] }: Collection = currentCollection;
     const isFolder: boolean = view === VIEW_FOLDER;
     const showSort: boolean = isFolder && items.length > 0;
     const showAdd: boolean = (!!canUpload || !!canCreateNewFolder) && isFolder;
+    const showGridButton: boolean = isFolder && items.length > 0;
```

Comment:
Should the `items.length > 0` check move to its own local variable, `hasItems`, since it's used more than once?